### PR TITLE
fetch common content

### PR DESCRIPTION
**Pull request by bnchdrff**
see #22

### CODE REVIEW

**Comment by stefanjudis** [comment]
@bnchdrff Nice!!! Thank you very much.

Hopefully I'll make it to have a look later today. :)

**Comment by bnchdrff** [comment]
writing docs now...

File: docs/global-settings.md [hunk 1]
```diff
@@ -81,3 +81,25 @@ contentful:
...
+For example, find your five latest entries so you can include them in various templates:
+
+```javascript
+{
+  // ...
+  "common": {
+    "latest": {
+      "content_type": "post",
+      "limit": 5,
+      "order": "sys.createdAt:
```
Comment (stefanjudis) [inline docs/global-settings.md]: `"` missing. :)

File: lib/processor.js [hunk 2]
```diff
@@ -121,6 +168,7 @@ function processFile (file, options) {
   return client.getEntries(query)
     .then(entries => mapEntriesForFile(entries.items, file, options))
     .then(entries => processEntriesForFile(file, entries))
+    .then(entries => getCommonContentForSpace(options, entries))
```
Comment (stefanjudis) [inline lib/processor.js]: Just a preference, but options feels better being one of the last parameters?

File: lib/processor.js [hunk 3]
```diff
@@ -26,6 +27,52 @@ function getContentfulClient (accessToken, spaceId, host) {
...
+ * @return {Object}        file mapping object with common content added
+ */
+function getCommonContentForSpace (options, entries) {
+  if (!options.common) {
+    return entries
+  }
+
+  const spaceId = options.space_id
+  const accessToken = options.access_token
+  const host = options.host
```
Comment (stefanjudis) [inline lib/processor.js]: Not a big deal, but...

Could we maybe create a helper function to help with `spaceId`, `accessToken` and `host`...
We've got 3 almost similar lines here as below. 

```
const spaceId = options.space_id
const accessToken = options.access_token
const host = options.host
```

Or we could move this logic into `getContentfulClient` and change the params?
I'm thinking of s.th. like

`getContentfulClient( options, file )`

which deals with the fallback logic?

Comment (bnchdrff) [inline lib/processor.js]: i agree that we should abstract this a bit... i'll work on it.

File: docs/global-settings.md (hunk 1 above)
Comment (bnchdrff) [inline docs/global-settings.md]: oops yup i'll get that !

File: lib/processor.js (hunk 2 above)
Comment (bnchdrff) [inline lib/processor.js]: sure that makes sense

File: docs/global-settings.md [hunk 4]
```diff
@@ -81,3 +81,25 @@ contentful:
...
+    }
+  },
+  // ...
+}
+```
+
+In templates you can then access `common.latest` to get the raw results of the above query.
```
Comment (stefanjudis) [inline docs/global-settings.md]: an example here would be great...

File: lib/processor.js (hunk 3 above)
Comment (bnchdrff) [inline lib/processor.js]: i'm just going to use the options direclty in my call to `getContentfulClient` for now -- still think you're right about refactoring this though.

**Comment by bnchdrff** [comment]
alright, changed per your review. thanks for lookin.

File: docs/global-settings.md [hunk 5]
```diff
@@ -81,3 +81,29 @@ contentful:
...
+    }
+  },
+  // ...
+}
+```
+
+In templates you can then access `common.latest` to get the raw results of the above query:
+
+```
+The first latest post's title: {{ common.latest.items.0.fields.title }}
```
Comment (bnchdrff) [inline docs/global-settings.md]: not in handlebars land 👨 !

i'll make a note in the example that it's a handlebars template.

File: lib/processor.js (hunk 3 above)
Comment (stefanjudis) [inline lib/processor.js]: Fair enough. :+1:

**Comment by stefanjudis** [comment]
@bnchdrff Cool cool cool.

I'll make a test round tomorrow morning (Berlin time). ;) 

Thx. :)

**Comment by stefanjudis** [comment]
Alright I checked it -> https://github.com/contentful-labs/contentful-metalsmith-example/pull/5

Looks good. Going to 🚢  it soon. 🎉

**Comment by bnchdrff** [comment]
@stefanjudis anonyben at gmail

**Pull request merged**